### ISSUE TITLE
Temporary bug - Derecho with RDMC broken in master

### ISSUE THREAD

**Issue by sagarjha** [Derecho-Project/derecho-unified#29]
It seems that some recent commits have broken test cases where Derecho sends data through RDMC. Hopefully, an easy fix.
I tested `derecho_bw_test` with 100 MB message size and I see a segfault because the buffer of the message accessed in the `version_message` function in `multicast_group.cpp` is `0x0`.

**Comment by sagarjha** [comment]
Fixed.
Derecho development is a distributed process with each one of us adding code. Just like any distributed system, an individual lacks the knowledge of the entire state. This leads to bugs (inconsistencies) when someone edits code written by someone else.
I had helped Weijia timestamp messages after delivery for making a version. This versioning is done after the message is delivered. So in case of RDMC messages, the message buffer has been freed (moved) after delivery so that new messages can be sent. The versioning needs to read this buffer to get the message timestamp. So I had made sure that we get the timestamp before delivery and use it after delivery for versioning. Edward refactored this code last month and moved the versioning to a new function. He moved the logic to retrieve the timestamp to that function and called it after deliver_message. This led to segmentation fault because the memory had been moved. He was only testing the code path that used SST so he did not notice the bug.
I have fixed this and tested successfully. Shortly, a commit will close this issue.
